lib/puppetserver/ca/action/prune.rb

```diff
@@ -50,40 +50,46 @@ def run(inputs)
           loader = X509Loader.new(puppet.settings[:cacert], puppet.settings[:cakey], puppet.settings[:cacrl])
 
           puppet_crl = loader.crls.select { |crl| crl.verify(loader.key) }
-          number_of_removed_duplicates = prune_CRLs(puppet_crl)
 
-          if number_of_removed_duplicates > 0
-            update_pruned_CRL(puppet_crl, loader.key)
-            FileSystem.write_file(puppet.settings[:cacrl], loader.crls, 0644)
-            @logger.inform("Removed #{number_of_removed_duplicates} duplicated certs from Puppet's CRL.")
+          if puppet_crl.length == 1
+            number_of_removed_duplicates = prune_CRL(puppet_crl.first)
+            @logger.inform("Total number of certificates found in Puppet's CRL is: #{puppet_crl.first.revoked.length}.")
```

Comment:
Curious about the `puppet_crl.first.revoked.length`, too much?

Reply:
I don't think it's too much, but if you wanted, you could change `puppet_crl` to something like `verified_crls` and then set a new variable `puppet_crl = verified_crls.first`, since you pass that to `prune_CRL` anyway.